### PR TITLE
CollectionView - Empty view for non-templated Header and footers

### LINKED ISSUE #8228: [Bug]Grouped CollectionView shows the item source object name in each group
### Description
I am using Grouping with CollectionView. The Collection View shows the name of the itemssource object in the end of every group. The name of the object in this case happens to be ShellRepro.CategoryGroup.

- Version with issue:Xamarin.Forms-4.3.0.908675
- Affected Devices:iOS

### Screenshots
CollectionView

### Reproduction Link
[ShellRepro.zip](https://github.com/xamarin/Xamarin.Forms/files/3772267/ShellRepro.zip)

### PR DESCRIPTION
**Pull request by memsranga**
### Description of Change ###
Replaced the Default Group Header and Footer with empty Views when no template is provided

### Issues Resolved ### 
 - fixes #8228 

### API Changes ###
 None

### Platforms Affected ###
- iOS

### Behavioral/Visual Changes ###
On iOS, when there is no template, removed the default Label (that was added with `ToString` on the bound object)

### Before/After Screenshots ### 
Before                     |  After
:-------------------------:|:-------------------------:
before | after

### Testing Procedure ###
Created a sample page with Grouped CollectionView with no Footer Template
Test Cases -> G8228

### PR Checklist ###
- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by shanranm** [comment]
The failing tests are related to CollectionView

File: Xamarin.Forms.Platform.iOS/CollectionView/GroupableItemsViewController.cs [hunk 1]
```diff
@@ -122,10 +120,7 @@ string DetermineViewReuseId(DataTemplate template)
 		{
 			if (template == null)
 			{
-				// No template, fall back the the default supplemental views
-				return ItemsViewLayout.ScrollDirection == UICollectionViewScrollDirection.Horizontal
-					? HorizontalDefaultSupplementalView.ReuseId
-					: VerticalDefaultSupplementalView.ReuseId;
+				return EmptySupplementalView.ReuseId;
```
Comment (jfversluis) [inline Xamarin.Forms.Platform.iOS/CollectionView/GroupableItemsViewController.cs]: I see you removed the whole horizontal/vertical logic. How does this behave when in horizontal mode?

**Review by jfversluis — COMMENTED** [review]
> The failing tests are not related to CollectionView

Actually there is at least one that is related to CollectionView. But it doesn't seem related to this change.

File: Xamarin.Forms.Platform.iOS/CollectionView/GroupableItemsViewController.cs (hunk 1 above)
Comment (shanranm) [reply Xamarin.Forms.Platform.iOS/CollectionView/GroupableItemsViewController.cs]: Same applies for horizontal mode... Attaching screentshots, see the `Xamarin.Forms.Controls.Issues.CategoryGroup` at center of screen, thats the current default footer
### Horizontal mode Before/After Screenshots ### 
Before                     |  After
:-------------------------:|:-------------------------:
before | after

**Comment by shanranm** [comment]
@jfversluis any update on this?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue8228.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 8228, "[Bug]Grouped CollectionView shows the item source object name in each group", PlatformAffected.iOS)]
+	public class Issue8228 : TestContentPage
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue8228.cs]: This file is not added to the `Xamarin.Forms.Controls.Issues.Shared.shproj` and doesn't show up in the gallery app

Comment (jfversluis) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue8228.cs]: Also, would it be possible to add a UITest? Just a simple one that navigates here and checks that there is no ToString value of sorts

**Review by jfversluis — CHANGES_REQUESTED** [review]

**Comment by samhouts** [comment]
@shanranm Are you able to address the comments? Thanks!

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue8228.cs (hunk 2 above)
Comment (shanranm) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue8228.cs]: Sorry for the late response!!
Included the file in shproj and added test to test to see if empty ones doesnt show up

**Comment by jfversluis** [comment]
I think at this point it's safe to say that we won't be taking this for Xamarin.Forms anymore. Thank you all so much for your time and effort on this. It hasn't gone unnoticed!

**Pull request closed without merging**